### PR TITLE
Feature/73 changes

### PR DESCRIPTION
**Pull request by cookersjs**
Purely Jenkinsfile PR
- imports graph database _with_ interactions, sets it up on release server
- generates analysis-core file and symlinks it to analysis.bin
- Archives everything, stores it on S3, performs cleanup

### CODE REVIEW

File: Jenkinsfile
Comment (SolomonShorser-OICR) [inline Jenkinsfile:70]: This is not the only place where the graph database gets generated. Is it possible to move this part of the Step to some sort of common library so that multiple steps can use the same code for generating the graph database?

Comment (cookersjs) [reply Jenkinsfile:70]: Yea I realize that would be better, it could be consolidated into a commonly used graph-database-importer jenkinsfile in release-jenkins-utils. It could also do automated things like reset neo4j and run graph-qa. I just wanted to write everything out since this is the first release that this is being done. If we want to leave this PR open until I do that, thats fine, though I'd like to use this for this release since I know it works

File: Jenkinsfile
Comment (SolomonShorser-OICR) [inline Jenkinsfile:168]: This will always pull... master or develop? I guess whatever the default branch is. Is it worth considering `git checkout $REPO_VERSION`?

File: Jenkinsfile
Comment (SolomonShorser-OICR) [reply Jenkinsfile:70]: No that's OK. I mostly wanted to just raise it as a discussion point. You could create an issue to track progress on that task, but this is not a blocker for now.

File: Jenkinsfile
Comment (cookersjs) [reply Jenkinsfile:168]: develop is the default branch. I think that that will be good practice when Jenkins isn't using such fresh branches, once we move into stable branches with set releases

Comment (SolomonShorser-OICR) [reply Jenkinsfile:168]: Yeah, not something to change for this PR, but to keep in mind for the future.

**Review by SolomonShorser-OICR — DISMISSED** [review]

File: Jenkinsfile
Comment (jweiser) [inline Jenkinsfile:44]: Similar to analysis-core PR (https://github.com/reactome/release-jenkins-utils/pull/3#discussion_r436200985) - maybe `${env.REACTOME_DB}` in place of `${env.REACTOME}`

File: Jenkinsfile
Comment (jweiser) [inline Jenkinsfile:56]: Similar to https://github.com/reactome/release-jenkins-utils/pull/3#discussion_r436122019 - including the slash separator in the paths may be easier to read than in the variable

File: Jenkinsfile
Comment (jweiser) [reply Jenkinsfile:70]: I agree with @SolomonShorser-OICR - creating an issue for tracking would be good but no need to block this PR.

File: Jenkinsfile
Comment (jweiser) [inline Jenkinsfile:143]: This method's body is fairly long - maybe you could add helper functions that group lines that are doing similar logic.  Like here, a function that is called to move the logs into their "logs" destination folder.

That way the body of the function would be shorter and the operations would be more English like and readable at a glance:

```
archiveFilesToLogFolder()
archiveFilesToDataFolder()
uploadFoldersToS3()
removeLocalFolders()
```

File: Jenkinsfile
Comment (jweiser) [inline Jenkinsfile:145]: A function to move data's contents?

File: Jenkinsfile
Comment (jweiser) [inline Jenkinsfile:146]: A function to move data's contents?

File: Jenkinsfile
Comment (jweiser) [inline Jenkinsfile:147]: Have gzip being called at the end of the functions which move content into "data" and "logs", respectively?

File: Jenkinsfile
Comment (jweiser) [inline Jenkinsfile:149]: Could move into the function responsible for moving files into "data" - either after gzipping or using a more specific pattern with gzip to avoid zipping these files.

File: Jenkinsfile
Comment (jweiser) [inline Jenkinsfile:156]: Move to a clean-up/deletion function?  

Also, the rm -rf statements could be consolidated: `rm -rf graph-importer* analysis-core* release-jenkins-utils*`

File: Jenkinsfile
Comment (jweiser) [inline Jenkinsfile:152]: Function to copy these directories to S3?

File: Jenkinsfile
Comment (jweiser) [reply Jenkinsfile:168]: I think `git pull` will pull everything for the repo regardless of the branch, but you need to do an explicit `git checkout [branch/tag]` if you aren't already on the branch you want.

File: Jenkinsfile
Comment (jweiser) [inline Jenkinsfile:163]: Similar to https://github.com/reactome/release-jenkins-utils/pull/3#discussion_r436201736 - this function will be useful across the Jenkins Release Pipeline.  If it can be moved to a new utilitiy file that is agnostic of any particular release step and imported where needed, it would reduce the duplication across all the Jenkins steps/modules.

**Review by jweiser — COMMENTED** [review]
No blockers for me, but I left some suggestions for feedback before merging.

**Review by SolomonShorser-OICR — APPROVED** [review]

**Comment by cookersjs** [comment]
All suggestions in this PR have been noted in issues or my personal notes (August 19th, 2020). They will be implemented or formally noted (via issues) ASAP.

**Pull request merged**
